### ISSUE TITLE
Load data from file in JavaScript

### ISSUE THREAD

**Issue by ddeboer** [oxigraph/oxigraph#906]
**Is your feature request related to a problem? Please describe.**

The JS API allows to `Store.load()` a string, but that doesn’t work well for large RDF files.

**Describe the solution you'd like**

Expose an (e.g.) `Store.loadFIle(filename)` function that loads from a file similarly to `oxigraph load --file filename.nt`.

**Describe alternatives you've considered**

* Load the file in memory first, then feed that to `Store.load()`. Doesn’t work for RDF files that don’t fit into memory and is inefficient in JS.
* Parse the file into quads in JS and pass that to `Store.add(quad)`. Too slow for large datasets.

**Additional context**

**Comment by Tpt** [comment]
Thank you for this feature request. Sadly Oxigraph for JavaScript is compiled for both web and nodeJS, and there is no shared API to access the file system between the two. What I am considering instead is making the load function take a [ReadableStream](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream) object instead. This way it would allow to load files, http responses...

**Comment by ddeboer** [comment]
Yes, taking a stream would be the way to go. As an aside, I just saw #628 which talks about this.

**Comment by minorg** [comment]
@Tpt How much work do you think this is? Does the Rust already use streams? https://github.com/MattiasBuelens/wasm-streams

**Comment by Tpt** [comment]
The Rust parsers can read [`AsyncRead`](https://docs.rs/tokio/latest/tokio/io/trait.AsyncRead.html) behind the `async-tokio` feature. However, note that this Rust library is for web streams and not NodeJS streams that the RDF/JS specs relies on. It will be a bit of work to make all of this work together.
